src/CommunityToolkit.Maui/Categories/ScreenCategorizer.shared.cs

```diff
@@ -0,0 +1,75 @@
+﻿using CommunityToolkit.Maui.Core.Primitives;
+using CommunityToolkit.Maui.Extensions;
+using CommunityToolkit.Maui.Mappings;
+using CommunityToolkit.Maui.Helpers;
+
+namespace CommunityToolkit.Maui.Categories;
+
+/// <summary>
+/// Responsible for categorizing the device's actual screen size to <see cref="ScreenCategories"/> based on <see cref="OnScreenSizeManager.Mappings"/>
+/// </summary>
+ class ScreenCategorizer : IScreenCategorizer
+{
+	public ScreenCategories GetCategoryByDiagonalSize(List<SizeMappingInfo> mappings, double deviceActualDiagonalSize)
+	{
+		ScreenCategories category;
+		if (TryCategorizeByFixedSize(mappings, deviceActualDiagonalSize, out category))
+		{
+			return category;
+		}
+
+		if (TryCategorizeBySmallerOrEqualsTo(mappings, deviceActualDiagonalSize, out category))
+		{
+			return category;
+		}
+
+		return ScreenCategories.NotSet;
+	}
+
+	/// <summary>
+	/// Attempt to categorize a screen based on fixed mapping screen size. 
+	/// </summary>
+	/// <param name="mappings"></param>
+	/// <param name="deviceActualDiagonalSize"></param>
+	/// <param name="category"></param>
+	/// <returns></returns>
+	 static bool TryCategorizeByFixedSize(List<SizeMappingInfo> mappings, double deviceActualDiagonalSize, out ScreenCategories category)
+	{
+		category = ScreenCategories.NotSet;
+
+		var diagonalSizeMappingsEquals = mappings.Where((f => f.ComparisonMode == ScreenSizeCompareModes.SpecificSize)).OrderBy(f => f.DiagonalSize).ToArray();
+		for (var index = 0; index < diagonalSizeMappingsEquals.Length; index++)
+		{
+			var sizeInfo = diagonalSizeMappingsEquals[index];
+			if (deviceActualDiagonalSize.EqualsTo(sizeInfo.DiagonalSize))
+			{
+				category = sizeInfo.Category;
+				return true;
+			}
+		}
```

Comment:
We don't need to create a concrete collection here. We can just use the `IEnumerable` instead.
```suggestion
var diagonalSizeMappingsEquals = mappings.Where((f => f.ComparisonMode == ScreenSizeCompareModes.SpecificSize)).OrderBy(f => f.DiagonalSize);

foreach (var sizeInfo in diagonalSizeMappingsEquals)
{
	if (deviceActualDiagonalSize.EqualsTo(sizeInfo.DiagonalSize))
	{
		category = sizeInfo.Category;
		return true;
	}
}
```

Reply:
Change a for loop, by a for each loop? 

In terms of performance a for loop is performant than a foreach loop. 

But I agree with you that in terms of memory allocation an ienumerable is better than a list